### ISSUE TITLE
The argument methodNameStyle of service macro should default to "Unchanged"

### ISSUE THREAD

**Issue by lilac** [higherkindness/sbt-mu-srcgen#93]
By default, when the "muSrcGenIdiomaticEndpoints" setting is true, the generated service has "methodNameStyle = Capitalize", this is a surprise to most grpc developers.

In addition, if muSrcGenIdiomaticEndpoints = false, the generated service does not have namespace set. Thus, there is not option available that allows us to have namespace and unchanged method names.

When we use protobuf as a rpc protocol, we often use some protobuf tools as well, like grpcui, or BloomRPC to test grpc services. However, due to the fact that grpc services generated by mu-srcgen is not compatible to its proto file, those tools do not work anymore.

In conclusion, we expect the generated grpc service to be like the following, to comply with conventions of grpc ecosystem.
```scala
@service(Protobuf, Identity, namespace = Some("example.seed.protocol.proto"), methodNameStyle = Unchanged) trait PeopleService[F[_]]
```

**Comment by dmarticus** [comment]
Hi @lilac, thanks for filing this issue!  This is an interesting request, I was under the impression that idiomatic gRPC services being capitalized _was_ the expected behavior for those services; in fact there was [an issue](https://github.com/higherkindness/mu-scala/issues/599) filed about a year ago about this very topic, and based on that discussion the assumption was that capitalizing the method names was how we should go about generating them.

That said, after digging around on the internet about it seems like your request is perfectly reasonable; we shouldn't necessarily mandate that all idiomatic endpoints have capitalized method names, and I think providing the option to make them unchanged while still providing the namespace and all that good stuff is something that mu could reasonably support.  I don't know if I'd be comfortable changing the default behavior (since changing defaults is a recipe for breaking things), but I'm definitely open to the idea of allowing mu users to specify both idiomatic endpoints _and_ unchanged method names.  Does that sound amenable to you?

**Comment by lilac** [comment]
@dmarticus Thanks for the reply. Having read through the issue you mentioned, I did not get a hint of why method names of an idiomatic endpoint should be capitalized. In addition, I think most grpc code generators and tools make the method names unchanged, just like you type in proto files. That's why I think it should be the default behavior of mu-srcgen. In addition, since this plugin has not reached v1.0, this surprising behavior should be rectified ASAP to avoid more technical debt.

However, if a break change is not expected at this stage, adding another option is acceptable to me.

**Comment by dmarticus** [comment]
@lilac My pleasure!  And yeah, I should've clarified that the discussion didn't explain _why_ the previous development decided to capitalize it, just that they _did_.  I'm more than open to seeing if they'd be willing to change the default since we're not at v1.0.0, but either way I definitely agree that this option should exist.  I'll ask a few of the original engineers about this, too.

@pepegar @fedefernandez @cb372 what do you think about this proposed change?

**Comment by fedefernandez** [comment]
I'm fine with that breaking change. We'd want to add the option to customize that option.
